src/frontend/Ast_to_Mir.ml

```diff
@@ -348,16 +395,13 @@ let rec check_decl decl_type decl_id decl_trans smeta adlevel =
       {expr= Var decl_id; emeta= {mtype; mloc= smeta; madlevel= adlevel}}
       smeta
   in
-  let constraint_str =
-    mkstring smeta (constraint_to_string decl_trans Check)
-  in
   let args = extract_constraint_args decl_trans in
   match decl_trans with
   | Identity | Offset _ | Multiplier _ | OffsetMultiplier (_, _) -> []
   | LowerUpper (lb, ub) ->
       check_decl decl_type decl_id (Ast.Lower lb) smeta adlevel
       @ check_decl decl_type decl_id (Ast.Upper ub) smeta adlevel
-  | _ -> [chk constraint_str args]
+  | _ -> [chk (mkstring smeta (constraint_to_string decl_trans Check)) args]
```

Comment:
This is the fix for the LowerUpper thing - constraint_to_string was being called when we were expecting LowerUpper to have been converted away already.